aws-traffic-mirroring/templates/vpc-mirroring.yaml

```diff
@@ -25,17 +25,29 @@ Resources:
       Tags:
         - { Key: Name, Value: !Sub '${AWS::StackName}-target' }
 
-  MirrorSession:
+  MirrorSession1:
     Type: AWS::EC2::TrafficMirrorSession
     Properties:
-      Description: !Sub Mirror traffic from ${SourceEni} to ${TargetEni}
-      NetworkInterfaceId: !Ref SourceEni
-      SessionNumber: 1
+      Description: Mirror traffic from !Select [0, !Ref SourceEniList ] to ${TargetEni}
+      NetworkInterfaceId: !Select [0, !Ref SourceEniList ] # get the relevant element by index (zero-based) from SourceEniList
+      SessionNumber: 1 # this should be unique for each MirrorSession object
       Tags:
         - { Key: Name, Value: !Sub '${AWS::StackName}-session' }
       TrafficMirrorFilterId: !Ref MirrorAlbTargetTraffic
       TrafficMirrorTargetId: !Ref MirrorTarget
-      VirtualNetworkId: 12345
+      VirtualNetworkId: 12345 # this should be unique for each MirrorSession object
```

Comment:
parameterized it?

Reply:
what do you mean? I can't dynamically calculate it